objstore.go

```diff
@@ -592,15 +593,25 @@ func (b *metricBucket) Upload(ctx context.Context, name string, r io.Reader) err
 	const op = OpUpload
 	b.ops.WithLabelValues(op).Inc()
 
-	start := time.Now()
-	if err := b.bkt.Upload(ctx, name, r); err != nil {
+	trc := newTimingReadCloser(
+		io.NopCloser(r),
```

Comment:
We don't use this pattern for `Get` and `GetRange`. What's the reason for treating `Upload` in a special way?

Reply:
I checked and we do use the TimingReadCloser in both `Get` and `GetRange`, (Lines 543 and 565)
Also threads above where this implementation was discussed: https://github.com/thanos-io/objstore/pull/66#discussion_r1280332799 and https://github.com/thanos-io/objstore/pull/66#discussion_r1280328066

Reply:
What I meant was pass in a `NopCloser` into the `TimingReadCloser`, and then we close the `TimingReadCloser` ourselves, but we don't do this for `Get` and `GetRange`.

Reply:
Actually nevermind, we don't return a reader from the `Upload` method, so I think this makes sense.